sdks/python/apache_beam/io/gcp/bigquery_io_metadata.py

```diff
@@ -64,6 +64,8 @@ def create_bigquery_io_metadata():
     # As we do not want a bad label to fail the BQ job.
     if _is_valid_cloud_label_value(dataflow_job_id):
       kwargs['beam_job_id'] = dataflow_job_id
+  if step_name:
+    kwargs['step_name'] = step_name
```

Comment:
This is likely to cause the BQ job to fail. The step_name would need to be sanitized. For the cloud label format
https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements


- Keys have a minimum length of 1 character and a maximum length of 63 characters, and cannot be empty. Values can be empty, and have a maximum length of 63 characters.
- Keys and values can contain only lowercase letters, numeric characters, underscores, and dashes. All characters must use UTF-8 encoding, and international characters are allowed.

Reply:
You're right that we need to sanitize. The step name would be something like `CustomerTransform/ReadFromBigquery/Read/_CustomBigQuertSouce`. I can add sanitization to it.

Reply:
I've added a sanitizer and a conditional append.